rsc/src/test/java/com/cloudera/livy/rsc/TestSparkClient.java

```diff
@@ -346,6 +337,48 @@ void config(Properties conf) {
   }
 
   @Test
+  public void testKillServerWhileSparkSubmitIsRunning() throws Exception {
+    Properties conf = createConf(true);
+    LivyClient client = null;
+    PipedInputStream stubStream = new PipedInputStream(new PipedOutputStream());
+    try {
+      Process mockSparkSubmit = mock(Process.class);
+      when(mockSparkSubmit.getInputStream()).thenReturn(stubStream);
+      when(mockSparkSubmit.getErrorStream()).thenReturn(stubStream);
+
+      // Block waitFor until process.destroy is called.
+      CountDownLatch waitForCalled = new CountDownLatch(1);
+      when(mockSparkSubmit.waitFor()).thenAnswer(new BlockingAnswer<>(waitForCalled, 1));
+
+      // Block waitFor until process.destroy is called.
+      CountDownLatch destroyCalled = new CountDownLatch(1);
+      Mockito.doAnswer(new BlockingAnswer<>(destroyCalled, null)).when(mockSparkSubmit).destroy();
+
+      ContextLauncher.mockSparkSubmit = mockSparkSubmit;
+      conf.put(TEST_MOCK_SPARK_SUBMIT.key(), "true");
+
+      client = new LivyClientBuilder(false).setURI(new URI("rsc:/"))
+        .setAll(conf)
+        .build();
+
+      client.stop(true);
+
+      waitForCalled.countDown();
+      destroyCalled.countDown();
```

Comment:
I'm confused, what is being tested now?

You want the `client.stop(true)` call to cause `Process.destroy()` to be called, right? So the answer in your mocked `destroy()` method should count down, and here you should instead wait for that to happen with a reasonable timeout.